### PR TITLE
FtE - Purity of spirit

### LINKED ISSUE #3143: FTE - Purity of Spirit

### PR DESCRIPTION
**Pull request by Matthew-Griffith**
closes #3143

### CODE REVIEW

File: server/game/cards/09.1-FtE/PurityOfSpirit.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                cardType: CardTypes.Character,
+                cardCondition: card => card.hasTrait('bushi') && card.isParticipating(),
+                gameAction: AbilityDsl.actions.multiple([
+                    AbilityDsl.actions.honor(),
+                    AbilityDsl.actions.delayedEffect({
+                        when : {
+                            onConflictFinished: () => true
+                        },
+                        gameAction: AbilityDsl.actions.discardStatusToken(context => ({ target: context.target.personalHonor })),
+                        message: 'remove the status token from {0}'
```
Comment (jamesoddy) [inline server/game/cards/09.1-FtE/PurityOfSpirit.js]: This isn't going to show a very good message in chat.  You probably want something like:
```
message: '{2} removes the {3} from {1} due to the delayed effect of {0}',
messageArgs: context => [context.player, context.target.personalHonor]
```

Comment (Matthew-Griffith) [reply server/game/cards/09.1-FtE/PurityOfSpirit.js]: so it isn't possible to pass in the `messageArgs` to the function. the interface to `DelayedEffectActionProperties` only defines when, message and gameAction. so I assume this was the intent when the delay effect action was made. I did change the message to 
`'the status token has been removed from {1} due to the delayed effect of {0}'`
let me know if  this is a good message given that the message will only get an array with 'Purity of Spirit' and the name of the character effected

File: server/game/cards/09.1-FtE/PurityOfSpirit.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                cardCondition: card => card.hasTrait('bushi') && card.isParticipating(),
+                gameAction: AbilityDsl.actions.multiple([
+                    AbilityDsl.actions.honor(),
+                    AbilityDsl.actions.delayedEffect({
+                        when : {
+                            onConflictFinished: () => true
+                        },
+                        gameAction: AbilityDsl.actions.discardStatusToken(context => ({ target: context.target.personalHonor })),
+                        message: '{2} removes the {3} from {1} due to the delayed effect of {0}',
+                        messageArgs: context => [context.player, context.target.personalHonor]
```
Comment (jeremylarner) [inline server/game/cards/09.1-FtE/PurityOfSpirit.js]: Have you checked to make sure this works in chat?  I don't think we've ever sent a token direct to chat before....

Comment (jamesoddy) [reply server/game/cards/09.1-FtE/PurityOfSpirit.js]: It's in the tests:
```
expect(this.getChatLogs(5)).toContain('player1 removes the Honored Token from Shiba Tsukune due to the delayed effect of Purity of Spirit');
```
Are we bothered about the capitalisation on "Honored Token"?

**Comment by jamesoddy** [comment]
Re-done on #3219

**Pull request merged**